master/buildbot/test/integration/test_latent.py

```diff
@@ -0,0 +1,224 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from buildbot.config import BuilderConfig
+from buildbot.process.factory import BuildFactory
+from buildbot.test.fake.reactor import NonThreadPool
+from buildbot.test.fake.reactor import TestReactor
+from buildbot.test.util.integration import getBuilderIdByName
+from buildbot.test.util.integration import getMaster
+from buildbot.worker.base import AbstractLatentWorker
+
+from twisted.internet.defer import Deferred
+from twisted.python import threadpool
+from twisted.python.failure import Failure
+from twisted.trial.unittest import SynchronousTestCase
+
+
+class LatentController(object):
+    """
+    A controller for ``ControllableLatentWorker``.
```

Comment:
I think this makes sense to add this as a in-code comment.

That blog post is great